### PR TITLE
Add new funtionality to search file that is Hidden readOnly

### PR DESCRIPTION
**Pull request by TheYisusGithub**

### CODE REVIEW

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 1]
```diff
@@ -17,49 +18,75 @@
...
-  * @author Jesus Menacho
-  * @version 1.0 *
-  */
-public class SearchVideo implements ISearch {
+/** Implements the model class File and the getter and setter´s methods
+* @author Jesus Menacho
+* @version 1.0 *
+*/
+public class SearchVideo implements ISearch{
 static Criteria criteria;
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: add the corresponding access modifier.

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 2]
```diff
@@ -17,49 +18,75 @@
...
                 }
             }
-            if (!(criteria.getPath().equals("")) && (criteria.getFileName().equals("")) && (!criteria.getExtension().equals(""))) {
-                if (fileExtractor.getName().contains(criteria.getExtension())) {
-                    listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
+            else
+            {
+                if ((fileExtractor.canRead() && !fileExtractor.canWrite()) == criteria.getFileReadOnly())
+                {
+                    if (compareFile(fileExtractor)!=null){
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: space before and after the logic operator "!="

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 3]
```diff
@@ -17,49 +18,75 @@
...
+                    listFileAndDirectory.add(compareFile(fileExtractor));
+                    }
                 }
-            }
-            if (!(criteria.getPath().equals("")) && (!criteria.getFileName().equals("")) && (!criteria.getExtension().equals(""))) {
-                if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getName().contains(criteria.getExtension())) {
-                    listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
+                else
+                {
+                    if (compareFile(fileExtractor)!=null){
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: space before and after the logic operator "!="

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 4]
```diff
@@ -17,49 +18,75 @@
...
+            else
+            {
+                if ((fileExtractor.canRead() && !fileExtractor.canWrite()) == criteria.getFileReadOnly())
+                {
+                    if (compareFile(fileExtractor)!=null){
+                    listFileAndDirectory.add(compareFile(fileExtractor));
+                    }
+                }
+                else
+                {   if (compareFile(fileExtractor)!=null){
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: space before and after the logic operator "!="

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 5]
```diff
@@ -17,49 +18,75 @@
...
+            if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getPath().contains(criteria.getPath())) {
+                return (new File(fileExtractor.getAbsolutePath()));
+            }
+        }
+        if (!(criteria.getPath() == null) && (!(criteria.getFileName() == null)) && (!(criteria.getExtension() == null))) {
+            if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getName().contains(criteria.getExtension())) {
+                return (new File(fileExtractor.getAbsolutePath()));
+            }
+        }
+        if (!(criteria.getPath() == (null)) && (criteria.getFileName()==null) && (!(criteria.getExtension() == null))) {
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: space before and after the logic operator "=="

**Review by aburgosu — CHANGES_REQUESTED** [review]

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 6]
```diff
@@ -1,13 +1,14 @@
- /**
-  * Copyright (c) 2019 Jalasoft.
-  * This software is the confidential and proprietary information of Jalasoft.
-  * ("Confidential Information"). You shall not
-  * disclose such Confidential Information and shall use it only in
-  * accordance with the terms of the license agreement you entered into
-  * with Jalasoft.
-  */
+/**
+*
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/model/SearchVideo.java:3]: Please fix the copyright tag according on given example

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 1 above)
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: add indentation

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 7]
```diff
@@ -17,49 +18,75 @@
...
-                listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
-            }
-            if (!(criteria.getPath().equals("")) && (!criteria.getFileName().equals("")) && (criteria.getExtension().equals(""))) {
-                if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getAbsolutePath().contains(criteria.getPath())) {
-                    listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
+            if (fileExtractor.isHidden() == criteria.getFileHidden())
+            {
+                if ((fileExtractor.canRead() && !fileExtractor.canWrite()) == criteria.getFileReadOnly())
+                {
+                    if (compareFile(fileExtractor)!=null){
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: space before and after the logic operator "!="

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 8]
```diff
@@ -17,49 +18,75 @@
...
+                    }
                 }
             }
         }
         return listFileAndDirectory;
     }
-}
-
+    private File compareFile(File fileExtractor){
+        if (!(criteria.getPath()==null) && (criteria.getFileName() == null) && (criteria.getExtension() == null)) {
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: spaces before and after "=="

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 9]
```diff
@@ -17,49 +18,75 @@
...
+            }
+        }
+        if (!(criteria.getPath() == (null)) && (criteria.getFileName()==null) && (!(criteria.getExtension() == null))) {
+            if (fileExtractor.getName().contains(criteria.getExtension())) {
+                return (new File(fileExtractor.getAbsolutePath()));
+            }
+        }
+        return null;
+    }
+}
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: new empty line at the end

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 10]
```diff
@@ -17,49 +18,75 @@
...
+* @version 1.0 *
+*/
+public class SearchVideo implements ISearch{
 static Criteria criteria;
-
     public SearchVideo(Criteria criteria){
         this.criteria = criteria;
     }
-
     public List searchByPath() {
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: method name should be search() as it it shown in ISearch interface

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 1 above)
Comment (TheYisusGithub) [reply src/main/java/com/fundation/logic/model/SearchVideo.java]: ok Andres that was correct.

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 2 above)
Comment (TheYisusGithub) [reply src/main/java/com/fundation/logic/model/SearchVideo.java]: oks. this was fixed

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 4 above)
Comment (TheYisusGithub) [reply src/main/java/com/fundation/logic/model/SearchVideo.java]: that was fixed.

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 5 above)
Comment (TheYisusGithub) [reply src/main/java/com/fundation/logic/model/SearchVideo.java]: that was fixed.

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 6 above)
Comment (TheYisusGithub) [reply src/main/java/com/fundation/logic/model/SearchVideo.java:3]: * that was remove  them that was fixed.

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 1 above)
Comment (TheYisusGithub) [reply src/main/java/com/fundation/logic/model/SearchVideo.java]: the indentations was apply.

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 7 above)
Comment (TheYisusGithub) [reply src/main/java/com/fundation/logic/model/SearchVideo.java]: that was fixed.

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 8 above)
Comment (TheYisusGithub) [reply src/main/java/com/fundation/logic/model/SearchVideo.java]: that was fixed.

File: src/main/java/com/fundation/logic/model/SearchVideo.java (hunk 9 above)
Comment (TheYisusGithub) [reply src/main/java/com/fundation/logic/model/SearchVideo.java]: ok that was fixed.

File: src/main/java/com/fundation/logic/model/SearchVideo.java [hunk 11]
```diff
@@ -17,49 +17,75 @@
...
         File[] allSubFiles = file.listFiles();
         for (File fileExtractor : allSubFiles) {
-            if (!(criteria.getPath().equals("")) && (criteria.getFileName().equals("")) && (criteria.getExtension().equals(""))) {
-                listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
-            }
-            if (!(criteria.getPath().equals("")) && (!criteria.getFileName().equals("")) && (criteria.getExtension().equals(""))) {
-                if (fileExtractor.getName().contains(criteria.getFileName()) && fileExtractor.getAbsolutePath().contains(criteria.getPath())) {
-                    listFileAndDirectory.add(new File(fileExtractor.getAbsolutePath()));
+            if (fileExtractor.isHidden() == criteria.getFileHidden())
+            {
```
Comment (MelissaRoman06) [inline src/main/java/com/fundation/logic/model/SearchVideo.java]: { should be in the previous line. This apply for all the code.

**Pull request merged**